custom/abdm/discovery_request.py

```diff
@@ -0,0 +1,136 @@
+from corehq.form_processor.models import CommCareCase
+from corehq.toggles import ABDM_INTEGRATION
+from corehq.toggles.shortcuts import find_domains_with_toggle_enabled
+from corehq.apps.es.case_search import CaseSearchES, case_property_range_query, sounds_like_text_query
+from corehq.apps.es import queries
+
+from abdm_integrator.hip.exceptions import DiscoveryNoPatientFoundError, DiscoveryMultiplePatientsFoundError
+
+from datetime import date
+from abdm_integrator.const import IdentifierType
+
+PATIENT_CASE_TYPE = 'abdm_patient_discovery_data'
+HIP_ID_PROPERTY = 'hip_id'
+HEALTH_ID_PROPERTY = 'abha_address'
+PHONE_NUMBER_PROPERTY = 'phone_number'
+DATE_OF_BIRTH_PROPERTY = 'dob'
+GENDER_PROPERTY = 'sex'
+ABHA_NUMBER_PROPERTY = 'abha_id'
+NAME_PROPERTY = 'case_name'
+
+CARE_CONTEXT_CASE_TYPE = 'fhir_care_context'
+YEARS_RANGE_TO_MATCH = 2
+
+
+def base_discover_query(domains, hip_id):
+    return CaseSearchES().filter(
+        {
+            'terms': {
+                'domain.exact': domains
```

Comment:
Why not use `CaseSearchES().domain` filter?

Reply:
Addressed in 1c52a52dc8d6bdd0b0b55c8a8eecc7efe41014e6